src/corehost/common/pal.h

```diff
@@ -146,24 +146,25 @@ namespace pal
     inline int strncasecmp(const char_t* str1, const char_t* str2, int len) { return ::_wcsnicmp(str1, str2, len); }
 
     inline size_t strlen(const char_t* str) { return ::wcslen(str); }
-    inline FILE * file_open(const pal::string_t& path, const char_t* mode) { return ::_wfopen(path.c_str(), mode); }
+    inline FILE * file_open(const string_t& path, const char_t* mode) { return ::_wfopen(path.c_str(), mode); }
+
     inline void file_vprintf(FILE* f, const char_t* format, va_list vl) { ::vfwprintf(f, format, vl); ::fputwc(_X('\n'), f); }
     inline void err_fputs(const char_t* message) { ::fputws(message, stderr); ::fputwc(_X('\n'), stderr); }
     inline void out_vprintf(const char_t* format, va_list vl) { ::vfwprintf(stdout, format, vl); ::fputwc(_X('\n'), stdout); }
     inline int str_vprintf(char_t* buffer, size_t count, const char_t* format, va_list vl) { return ::_vsnwprintf(buffer, count, format, vl); }
 
-    bool pal_utf8string(const pal::string_t& str, std::vector<char>* out);
-    bool utf8_palstring(const std::string& str, pal::string_t* out);
-    bool pal_clrstring(const pal::string_t& str, std::vector<char>* out);
-    bool clr_palstring(const char* cstr, pal::string_t* out);
+    bool pal_utf8string(const string_t& str, std::vector<char>* out);
+    bool utf8_palstring(const std::string& str, string_t* out);
+    bool pal_clrstring(const string_t& str, std::vector<char>* out);
+    bool clr_palstring(const char* cstr, string_t* out);
 
-    inline bool mkdir(const pal::char_t* dir, int mode) { return CreateDirectoryW(dir, NULL) != 0; }
-    inline bool rmdir (const pal::char_t* path) { return RemoveDirectoryW(path) != 0; }
-    inline int rename(const pal::char_t* old_name, const pal::char_t* new_name) { return ::_wrename(old_name, new_name); }
-    inline int remove(const pal::char_t* path) { return ::_wremove(path); }
+    inline bool mkdir(const char_t* dir, int mode) { return CreateDirectoryW(dir, NULL) != 0; }
+    inline bool rmdir (const char_t* path) { return RemoveDirectoryW(path) != 0; }
+    inline int rename(const char_t* old_name, const char_t* new_name) { return ::_wrename(old_name, new_name); }
+    inline int remove(const char_t* path) { return ::_wremove(path); }
+    inline void unmap_file(void* addr, size_t length) { return UnmapViewOfFile(addr) != 0; }
```

Comment:
The function returns `void` - so it can't report a failure - also the body now contains a `return` in a function which is `void`.